examples/portworx/main.tf

```diff
@@ -0,0 +1,159 @@
+  manage_default_route_table    = true
+  default_route_table_tags      = { Name = "${local.name}-default" }
+  manage_default_security_group = true
+  default_security_group_tags   = { Name = "${local.name}-default" }
+
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/elb"                      = 1
+  }
+
+  private_subnet_tags = {
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/internal-elb"             = 1
+  }
+
+  tags = local.tags
+}
+
+#---------------------------------------------------------------
+# Custom IAM roles for Node Groups
+#---------------------------------------------------------------
+
+resource "aws_iam_policy" "portworx_eksblueprint_volume_access" {
+  name = "portworx_eksblueprint_volume_access"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "ec2:AttachVolume",
+          "ec2:ModifyVolume",
+          "ec2:DetachVolume",
+          "ec2:CreateTags",
+          "ec2:CreateVolume",
+          "ec2:DeleteTags",
+          "ec2:DeleteVolume",
+          "ec2:DescribeTags",
+          "ec2:DescribeVolumeAttribute",
+          "ec2:DescribeVolumesModifications",
+          "ec2:DescribeVolumeStatus",
+          "ec2:DescribeVolumes",
+          "ec2:DescribeInstances",
+          "autoscaling:DescribeAutoScalingGroups"
+        ]
+        Effect   = "Allow"
+        Resource = "*"
+      },
+    ]
+  })
+}
+
+
+#---------------------------------------------------------------
+# EKS Blueprints
+#---------------------------------------------------------------
+
+module "eks_blueprints" {
+  source = "../.."
+
+  cluster_name    = local.cluster_name
+  cluster_version = "1.22"
+
+  vpc_id             = module.vpc.vpc_id
+  private_subnet_ids = module.vpc.private_subnets
+
+  managed_node_groups = {
+    eksblueprint_nodegroup_med = {
+      node_group_name         = "eksblueprint_nodegroup_med"
+      instance_types          = ["t2.medium"]
+      min_size                = 3
+      desired_size            = 3
+      max_size                = 3
+      subnet_ids              = module.vpc.private_subnets
+      additional_iam_policies = [aws_iam_policy.portworx_eksblueprint_volume_access.arn]
+    }
+  }
+  tags = local.tags
+
+  depends_on = [
+    aws_iam_policy.portworx_eksblueprint_volume_access
+  ]
+}
+
+module "eks_blueprints_kubernetes_addons" {
+  source               = "../../modules/kubernetes-addons"
+  eks_cluster_id       = module.eks_blueprints.eks_cluster_id
+  eks_cluster_endpoint = module.eks_blueprints.eks_cluster_endpoint
+  eks_oidc_provider    = module.eks_blueprints.oidc_provider
+  eks_cluster_version  = module.eks_blueprints.eks_cluster_version
+
+  enable_portworx = true
+
+  portworx_chart_values = {
+    clusterName  = "portworx-example"
+    imageVersion = "2.11.2"
+  }
```

Comment:
```suggestion

  portworx_helm_config = {
    set_values = [
      {
        name = "clusterName"
        value = module.eks_blueprints.eks_cluster_id
        type  = string
      },
      {
        name  = "imageVersion"
        value = "2.11.2"
        type  = string
      }
    ]
  }
```

Reply:
Can you confirm that the field name is supposed to be "set_values" or "set".  ```helm_addon``` module does not support the "set_values" field but "set". Is there something that I am missing

Reply:
@askulkarni2  ^^

Reply:
@pragrawal10, apologies for not being clear on this. The `set_values` field is the one you would pass down to the `helm-addon` module (see https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/modules/kubernetes-addons/helm-addon/variables.tf)

Reply:
@askulkarni2 I made the changes that you have suggested, can you review it.